doc/sphinx/language/gallina-extensions.rst

```diff
@@ -951,83 +944,39 @@ Reserved commands inside an interactive module
-    to :token:`module_expression`.
-
-    .. cmdv:: Module @ident {* @module_binding} := @module_expression
-
-       Defines a functor with parameters given by the list of :token:`module_binding` and body :token:`module_expression`.
-
-    .. cmdv:: Module @ident {* @module_binding} : @module_type := @module_expression
-
-       Defines a functor with parameters given by the list of :token:`module_binding` (possibly none), and output module type :token:`module_type`,
-       with body :token:`module_expression`.
-
-    .. cmdv:: Module @ident {* @module_binding} <: {+<: @module_type} := @module_expression
-
-       Defines a functor with parameters given by module_bindings (possibly none) with body :token:`module_expression`.
-       The body is checked against each :n:`@module_type__i`.
-
-    .. cmdv:: Module @ident {* @module_binding} := {+<+ @module_expression}
-
-       is equivalent to an interactive module where each :token:`module_expression` is included.
-
-.. cmd:: Module Type @ident
-
-   This command is used to start an interactive module type :token:`ident`.
-
-   .. cmdv:: Module Type @ident {* @module_binding}
-
-      Starts an interactive functor type with parameters given by :token:`module_bindings`.
-
 
 Reserved commands inside an interactive module type:
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. cmd:: Include @module
-
-   Same as ``Include`` inside a module.
-
-.. cmd:: Include {+<+ @module}
-
-   This is a shortcut for the command :n:`Include @module` for each :token:`module`.
-
-.. cmd:: @assumption_token Inline @assums
-   :name: Inline
-
-   The instance of this assumption will be automatically expanded at functor application, except when
-   this functor application is prefixed by a ``!`` annotation.
-
-.. cmd:: End @ident
+.. cmdv:: End @ident
 
    This command closes the interactive module type :token:`ident`.
 
    .. exn:: This is not the last opened module type.
       :undocumented:
 
-.. cmd:: Module Type @ident := @module_type
+.. cmd:: Module Type @ident {* @module_binder } {+ <: @module_type_inl } {? @is_module_type }
 
-   Defines a module type :token:`ident` equal to :token:`module_type`.
+   .. insertprodn is_module_type is_module_type
 
-    .. cmdv:: Module Type @ident {* @module_binding} := @module_type
+   .. prodn::
+      is_module_type ::= := @module_type_inl {* <+ @module_type_inl }
 
-       Defines a functor type :token:`ident` specifying functors taking arguments :token:`module_bindings` and
-       returning :token:`module_type`.
+   This command is used to start an interactive module type :n:`@ident`.
 
-    .. cmdv:: Module Type @ident {* @module_binding} := {+<+ @module_type }
+   Specifying :n:`{* @module_binder }` starts an interactive functor type with
+   parameters given by the :n:`@module_binder`\s.
 
-       is equivalent to an interactive module type were each :token:`module_type` is included.
```

Comment:
This was forgotten in the description of `Module Type`.

Reply:
Same idea; see previous comment https://github.com/coq/coq/pull/11705#discussion_r396039668

Reply:
IIUC `Include` and `Include Type` in an interactive definition are equivalent to giving the same definitions in "Module" and "Module Type" respectively.  How would `Declare Module` in an interactive definition fit into a single-line module definition?  Is it true that everything you can do interactively has an equivalent single-command definition?  From your earlier example:
```
Module Type contains_empty (empty_arg : empty_type).
Declare Module empty : empty_type.
End contains_empty.
```

I can leave a `todo` here if this is getting in too deep here for this PR.

Reply:
> IIUC `Include` and `Include Type` in an interactive definition are equivalent to giving the same definitions in "Module" and "Module Type" respectively

That may have been the idea originally but now they are strictly equivalent commands.

>  Is it true that everything you can do interactively has an equivalent single-command definition?

No, I don't think so. Not in the case of modules.